### PR TITLE
OCPBUGS-30124: do not set KAS OAuthMetadataFile with Authentication type OIDC

### PR DESCRIPTION
**Pull request by sjenning**
supercedes https://github.com/openshift/hypershift/pull/3668

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kas/config.go [hunk 1]
```diff
@@ -122,15 +122,17 @@ func generateConfig(p KubeAPIServerConfigParams) *kcpv1.KubeAPIServerConfig {
...
-			OAuthMetadataFile: cpath(kasVolumeOauthMetadata().Name, OauthMetadataConfigKey),
-		},
 		ConsolePublicURL:             p.ConsolePublicURL,
 		ImagePolicyConfig:            imagePolicyConfig(p.InternalRegistryHostName, p.ExternalRegistryHostNames),
 		ProjectConfig:                projectConfig(p.DefaultNodeSelector),
 		ServiceAccountPublicKeyFiles: []string{cpath(kasVolumeServiceAccountKey().Name, pki.ServiceSignerPublicKey)},
 		ServicesSubnet:               strings.Join(p.ServiceNetwork, ","),
 	}
+
+	if p.Authentication != nil && p.Authentication.Type == configv1.AuthenticationTypeOIDC {
```
Comment (deads2k) [inline control-plane-operator/controllers/hostedcontrolplane/kas/config.go:132]: I might have misunderstood, but I thought we only did this if the type was the internal OIDC or empty.

Comment (sjenning) [reply control-plane-operator/controllers/hostedcontrolplane/kas/config.go:132]: wow yes... the opposite of this please :man_facepalming:

**Pull request merged**